packages/server/src/api/todo/adx/account/password-reset.ts

```diff
@@ -0,0 +1,105 @@
+import jwt from 'jsonwebtoken'
+import { ServerConfig } from '../../../../config'
+import { User } from '../../../../db/user'
+import { Server } from '../../../../lexicon'
+import * as locals from '../../../../locals'
+
+const PASSWORD_RESET_SCOPE = 'todo.adx.resetAccountPassword'
+
+export default function (server: Server) {
+  server.todo.adx.requestAccountPasswordReset(
+    async (_params, input, _req, res) => {
+      const { db, mailer, config } = locals.get(res)
+      const { email } = input.body
+
+      // @TODO should multiple users be able to have the same email?
+      const table = db.db.getRepository(User)
+      const users = await table.findBy({ email })
+
+      for (const user of users) {
+        // By signing with the password hash, this jwt becomes invalid once the user changes their password.
+        // This allows us to ensure it's essentially single-use (combined with the jwt's short-livedness).
+        const token = jwt.sign(
+          {
+            sub: user.did,
+            scope: PASSWORD_RESET_SCOPE,
+          },
+          getSigningKey(user, config),
+          { expiresIn: '15mins' },
+        )
+
+        await mailer.sendResetPassword({ token }, { to: user.email })
+      }
+
+      return {
+        encoding: 'application/json',
+        body: {},
+      }
+    },
+  )
+
+  server.todo.adx.resetAccountPassword(async (_params, input, _req, res) => {
+    const { db, config } = locals.get(res)
+    const { token, password } = input.body
+
+    const tokenBody = jwt.decode(token)
```

Comment:
`verify` also does `decode` so this code could probably be simplified just a bit

`const payload = jwt.verify(...)`

Reply:
ohhh wait you need the `did` from the decoded payload to get the signing key for verification

ignore prev message^^ 😛